src/sqlfluff/rules/L036.py

```diff
@@ -104,8 +104,15 @@ def _eval_multiple_select_target_elements(self, select_targets_info, segment):
                 == select_target.pos_marker.working_line_no
             ):
                 # Find and delete any whitespace before the select target.
+                start_seg = select_targets_info.select_idx
+                # If any select modifier (e.g. distinct ) is present, start
+                # there rather than at the beginning.
+                modifier = segment.get_child("select_clause_modifier")
```

Comment:
👍  IIRC, this rule has two major "paths", one for `SELECT` statements with a single target, and one for `SELECT` statements with multiple targets. Do we need to fix a similar problem in the other code path?

Reply:
From reading the logic I don't *think* so, but I'll add a few edge cases to make sure.

Reply:
@barrywhart - right you are - it currently mangles this:
```
SELECT
      distinct a
    FROM my_table
```

Will address that too.